### ISSUE TITLE
edgedb instance link prompts for password twice

### ISSUE THREAD

**Issue by fmoor** [edgedb/edgedb-cli#608]
When the `--password` option is passed then `edgedb instance link` prompts twice for the password. It seems like the user should only be asked once.

- EdgeDB CLI Version: EdgeDB CLI 1.0.0-dev.677+cbc0abb
- OS Version: ubuntu 21.04

Steps to Reproduce:
```bash
$ edgedb -H 164.90.155.183 --password instance link do --trust-tls-cert
Password for 'edgedb':
Specify the port of the server [default: 5656]:
> 5656
Specify the database user [default: edgedb]:
> edgedb
Specify the database name [default: edgedb]:
> edgedb
Trusting unknown server certificate: SHA1:00458653258f28a712c6eee807ed95a4ed4311d0
Password for 'edgedb':
```

**Comment by tailhook** [comment]
Can reproduce it on v1.0.0-rc.4-33-g8cf5464 anymore.

**Comment by fmoor** [comment]
hm... I'm still seeing this

```bash
$ edgedb instance link --trust-tls-cert --host myhost --password aws
Password for 'edgedb':
Specify the port of the server [default: 5656]:
> 5656
Specify the database user [default: edgedb]:
> edgedb
Specify the database name [default: edgedb]:
> edgedb
Trusting unknown server certificate: SHA1:ba4de6580e962d7225a5cfebaefd894c9910175d
Password for 'edgedb':
Successfully linked to remote instance. To connect run:
  edgedb -I aws
$ edgedb --password instance link --trust-tls-cert --host myhost aws
Password for 'edgedb':
Specify the port of the server [default: 5656]:
> 5656
Specify the database user [default: edgedb]:
> edgedb
Specify the database name [default: edgedb]:
> edgedb
Trusting unknown server certificate: SHA1:ba4de6580e962d7225a5cfebaefd894c9910175d
Password for 'edgedb':
/home/fmoor/.config/edgedb/credentials/aws.json exists! Overwrite? (type `Yes`)
> Yes
Successfully linked to remote instance. To connect run:
  edgedb -I aws
$ edgedb --version
EdgeDB CLI 1.1.0-dev.738+9ef4c0e
```

**Comment by tailhook** [comment]
Can you try reproducing this again. I can't anymore and I think it's fixed by #969. But I'm not sure since I couldn't easily reproduce this last time too.

**Comment by fmoor** [comment]
I can't reproduce any more using `EdgeDB CLI 3.0.0+8b024db`.
